secretkeylabs/xverse-web-extension
# Restore Wallet Update

# PR Type


# What is the current behavior?
- wallet restoration relied on fetching the recent transaction activity of every derivation of the wallet up to 15 accounts


# What is the new behavior?
- Wallet Restoration now relies on a wallet-config file stored in Gaia
- loading active accounts now happens whenever you log in with your password instead of the home screen
- create account functionality has been moved to the core-lib

# Testing instructions:
This Pr affects three things:

the restore wallet flow
the change network flow (triggers restore)
the login flow (triggers restore)
Basically, the way we restore wallets has changed so the most important thing is to make sure that works correctly in the following cases:

1. restoring a newly generated wallet with no accounts from any app
2. restoring a wallet that was created using Hiro wallet that has accounts
3. restoring a wallet that was created by the extension that has accounts
4. restoring a wallet that was created by the Mobile app that has accounts
for 3,4: There is a migration step that needs to happen where if you weren't logged in when this update is deployed and you were only using xverse apps you will find the accounts missing this is intentional and the course of action here is to manually generate the accounts again and it should appear once you restore for a second time
